### DISCUSSION TITLE
vuepress-theme-hope:  You are not allowed to use plugin "vuepress-plugin-feed2" yourself in vuepress config file, why ?

### DISCUSSION THREAD

**Discussion by StefanSa** [vuepress-theme-hope/vuepress-theme-hope#2992] [Q&A] [answered]
Hi there,
get this warning:
`vuepress-theme-hope:  ✖ You are not allowed to use plugin "vuepress-plugin-feed2" yourself in vuepress config file. Set "plugin.feed" in theme options to customize it` .

project was created with:
`pnpm create vuepress-theme-hope my-docs`

config.ts
```
import { defineUserConfig } from "vuepress";
import theme from "./theme.js";
import { searchProPlugin } from "vuepress-plugin-search-pro";
import { feedPlugin } from "vuepress-plugin-feed2";


export default defineUserConfig({
  base: "/",

  locales: {
    "/": {
      lang: "en-US",
      title: "Docs Demo",
      description: "A docs demo for vuepress-theme-hope",
    },
    "/zh/": {
      lang: "zh-CN",
      title: "文档演示",
      description: "vuepress-theme-hope 的文档演示",
    },
  },

   theme,

  shouldPrefetch: false,
  
  plugins: [
    searchProPlugin({
      // index all contents
      indexContent: true,
      // add supports for category and tags
      customFields: [
        {
          getter: (page) => page.frontmatter.category,
          formatter: "Category: $content",
        },
        {
          getter: (page) => page.frontmatter.tag,
          formatter: "Tag: $content",
        },
      ],
    }),

    feedPlugin({
          hostname: "doc-it.test.local",
          atom: true,
          rss: true,
        }),

  ],
  
  // Enable it with pwa
  // shouldPrefetch: false,
});

```

thanks for any help.

**Comment by Mister-Hope** [comment]
So what is problem for you to understand that warning?

**Comment by StefanSa** [comment]
@Mister-Hope 你好
First of all, thank for your great work here.
I would like to understand if i can igonize this warning or turn it off, followed the example and  get this warning.
https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/main/demo/feed2/src/.vuepress/config.ts

**Reply by Mister-Hope** [reply]
Isn't the fxxking warning tell you what to do itself?

Use a translator if you can not understand those words.

**Reply by StefanSa** [reply]
Sorry @Mister-Hope  am a beginner /nope here.
i want to understand why i get the warning when i use  your example. (maybe your example is wrong)
Sure i can read the warning text, but i don't know because i'm really a beginner what / where i have to do here.
Can you please show me that ?

**Reply by Mister-Hope** [reply]
So what is your trouble understanding `Set "plugin.feed" in theme options to customize it`?

This is not how you should ssk questions in open source project. This is an exact guideline which tells you what to do. You NEED to provide more details, like:

- What's theme options.
- how can I set plugins.feed

I am not supposed to teach you everything here.

**Comment by Oshibuki** [comment]
> 你好呀， 得到这个警告： `vuepress-theme-hope: ✖ You are not allowed to use plugin "vuepress-plugin-feed2" yourself in vuepress config file. Set "plugin.feed" in theme options to customize it` .
> 
> 项目创建于： `pnpm create vuepress-theme-hope my-docs`
> 
> 配置.ts
> 
> ```
> import { defineUserConfig } from "vuepress";
> import theme from "./theme.js";
> import { searchProPlugin } from "vuepress-plugin-search-pro";
> import { feedPlugin } from "vuepress-plugin-feed2";
> 
> 
> export default defineUserConfig({
>   base: "/",
> 
>   locales: {
>     "/": {
>       lang: "en-US",
>       title: "Docs Demo",
>       description: "A docs demo for vuepress-theme-hope",
>     },
>     "/zh/": {
>       lang: "zh-CN",
>       title: "文档演示",
>       description: "vuepress-theme-hope 的文档演示",
>     },
>   },
> 
>    theme,
> 
>   shouldPrefetch: false,
>   
>   plugins: [
>     searchProPlugin({
>       // index all contents
>       indexContent: true,
>       // add supports for category and tags
>       customFields: [
>         {
>           getter: (page) => page.frontmatter.category,
>           formatter: "Category: $content",
>         },
>         {
>           getter: (page) => page.frontmatter.tag,
>           formatter: "Tag: $content",
>         },
>       ],
>     }),
> 
>     feedPlugin({
>           hostname: "doc-it.test.local",
>           atom: true,
>           rss: true,
>         }),
> 
>   ],
>   
>   // Enable it with pwa
>   // shouldPrefetch: false,
> });
> ```
> 
> 谢谢你的帮助。

https://theme-hope.vuejs.press/zh/config/plugins/feed.html

**Comment by fdevans** [comment, marked as answer]
https://github.com/rundeck/docs/blob/vuepress2-migration/docs/.vuepress/config.ts

This is a working config.ts using the Feed plugin with it placed in the proper location to avoid the warning.

For the Hope Theme plugins they are best configured within the theme definition rather than the vuepress plugins section.
